docs: update docusaurus, add metrics endpoint, add zero downtime update (#4376)
zitadel/zitadel: docs/docs/concepts/architecture/solution.md

![Multi-Cluster Architecture](/img/zitadel_multicluster_architecture.png)

## Zero Downtime Updates

Since an Identity system tends to be a critical piece of infrastructure, the "in place zero downtime update" is a well needed feature.

> ZITADEL is built in a way that version can be updated without downtime by just updating to a more recent version.

Reviewer: ```suggestion
ZITADEL is built in a way that upgrades can be executed without downtime by just updating to a more recent version.
```

The common update involves the following steps and do not need manual intervention of the operator:

- Keep the old version running

> - Deploy the version parallel to the old version

Reviewer: ```suggestion
- Deploy the version in parallel to the old version
```

- The new version will start ...
  - by updating databases schemas if needed
  - participate in the leader election for background jobs
- As soon as the new version is ready to accept traffic it will signal this on the readiness endpoint `/debug/ready` 
- At this point your network infrastructure can send traffic to the new version

> Customer who use [Kubernetes/Helm](../../guides/deploy/kubernetes) or serverless container services like Google Cloud Run can benefit from the fact the above process is automated.

Reviewer: ```suggestion
Users who use [Kubernetes/Helm](../../guides/deploy/kubernetes) or serverless container services like Google Cloud Run can benefit from the fact the above process is automated.
```
